ukat/mapping/diffusion.py

```diff
@@ -0,0 +1,414 @@
+"""
+Diffusion imaging module
+
+"""
+import os
+import nibabel as nib
+import numpy as np
+import warnings
+
+from dipy.core.gradients import gradient_table, unique_bvals_tolerance
+from dipy.reconst.dti import TensorModel
+from tqdm import tqdm
+
+
+def make_gradient_scheme(bvals, bvecs, normalize=True, one_bzero=True):
+    """Make gradient scheme from list of bvals and bvecs
+
+    Parameters
+    ----------
+    bvals : list
+        b-values in s/mm2
+    bvecs : list of lists
+        bvectors (e.g. [[0, 0, 1], [1, 0, 0]])
+    normalize : bool, optional (default True)
+        Rescales bvecs to have unit length
+    one_bzero : bool, optional (default True)
+        Ensures gradient scheme only includes one b=0 measurement
+        If this is True and bvals does not contain any b=0, a b=0 measurement
+        will be included at the start of the acquisition
+
+    Returns
+    -------
+    string
+        gradient scheme with one line per measurement/volume as follows:
+        bvec1_x   bvec1_y   bvec1_z   bval1
+        bvec2_x   bvec2_y   bvec2_z   bval2
+        ...
+        bvecN_x   bvecN_y   bvecN_z   bvalN
+
+    Notes
```

Comment:
Well done, these notes are probably the most important part of this function.

After reading this, I guess that a future task is to improve the DTI input call? For example, the `gtab` argument could accept this string as argument and convert it to DiPy table?

Reply:
This function was all Fabio, I just moved it around in the file.

Reply:
I remember now. Ok, we'll have to keep this in mind for the future. At least, the situation is documented. It may a problem/challenge at pre-processing level in the User Interface.